Source/SwiftLintFramework/Rules/Lint/CompilerProtocolInitRule.swift

```diff
@@ -1,38 +1,45 @@
 import Foundation
 import SourceKittenFramework
 
-public struct CompilerProtocolInitRule: ASTRule, ConfigurationProviderRule, AutomaticTestableRule {
+public struct CompilerProtocolInitRule: ASTRule, ConfigurationProviderRule {
     public var configuration = SeverityConfiguration(.warning)
 
     public init() {}
 
-    public static let description = RuleDescription(
-        identifier: "compiler_protocol_init",
-        name: "Compiler Protocol Init",
-        description: "The initializers declared in compiler protocols such as `ExpressibleByArrayLiteral` " +
-                     "shouldn't be called directly.",
-        kind: .lint,
-        nonTriggeringExamples: [
-            "let set: Set<Int> = [1, 2]\n",
-            "let set = Set(array)\n"
-        ],
-        triggeringExamples: [
-            "let set = ↓Set(arrayLiteral: 1, 2)\n",
-            "let set = ↓Set.init(arrayLiteral: 1, 2)\n"
-        ]
-    )
+    public static let description = specificDescription(for: "such as `ExpressibleByArrayLiteral`", isPlural: true)
+
+    private static func specificDescription(for violation: String, isPlural: Bool) -> RuleDescription {
+        return RuleDescription(
+            identifier: "compiler_protocol_init",
+            name: "Compiler Protocol Init",
+            description: "The initializers declared in compiler protocol\(isPlural ? "s" : "") \(violation) " +
+                         "shouldn't be called directly.",
+            kind: .lint,
+            nonTriggeringExamples: [
+                "let set: Set<Int> = [1, 2]\n",
+                "let set = Set(array)\n"
+            ],
+            triggeringExamples: [
+                "let set = ↓Set(arrayLiteral: 1, 2)\n",
+                "let set = ↓Set.init(arrayLiteral: 1, 2)\n"
+            ]
+        )
+    }
 
     public func validate(file: File, kind: SwiftExpressionKind,
                          dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
         return violationRanges(in: file, kind: kind, dictionary: dictionary).map {
-            StyleViolation(ruleDescription: type(of: self).description,
-                           severity: configuration.severity,
-                           location: Location(file: file, characterOffset: $0.location))
+            let (violation, range) = $0
+            return StyleViolation(
+                ruleDescription: type(of: self).specificDescription(for: violation.protocolName, isPlural: false),
+                severity: configuration.severity,
+                location: Location(file: file, characterOffset: range.location)
```

Comment:
`StyleViolation` has a `reason` parameter which could be used to set specify violation reason without rebuilding the entire RuleDescription (which is used for generating documentation, unit tests, etc...)

Reply:
Good point! Didn't know about this one, thanks!